sdk/core/azure-core-test/src/main/java/com/azure/core/test/policy/RecordNetworkCallPolicy.java

```diff
@@ -217,15 +222,35 @@ private void captureRequestHeaders(HttpHeaders requestHeaders, Map<String, Strin
         }
     }
 
-    private String redactUserDelegationKey(String content) {
-        if (!content.contains("UserDelegationKey")) {
-            return content;
+    private String redactSensitiveContent(String content) {
+        // replace all sensitive string key in the black list
+        for (String blackString : BLACK_LIST) {
+            if (content.contains(blackString)) {
+                switch (blackString) {
+                    case USER_DELEGATION_KEY:
+                        content = redactUserDelegationKey(content);
+                        break;
+                    case ACCESS_TOKEN:
```

Comment:
Should we write simple unit test for this?

Reply:
updated